HopsanGUI/Widgets/HcomWidget.cpp

```diff
@@ -551,6 +595,30 @@ void TerminalConsole::keyPressEvent(QKeyEvent *event)
     }
 }
 
+//! @brief Inserts a selected completion from autocompleter
+void TerminalConsole::insertCompletion(const QString& completion)
+{
+    QString text = completion;
+    if(text.endsWith("()"))
+        text.chop(1);
+
+    if (mpCompleter->widget() != this)
+        return;
+    QTextCursor tc = textCursor();
+    int extra = text.length() - mpCompleter->completionPrefix().length();
+    tc.movePosition(QTextCursor::Left);
+    tc.movePosition(QTextCursor::EndOfWord);
+    tc.insertText(text.right(extra));
+    setTextCursor(tc);
+}
+
+//! @brief Updates list of auto complete words
+void TerminalConsole::updateAutoCompleteList()
+{
+    mpCompleter->setModel(new QStringListModel(mpTerminal->mpHandler->getAutoCompleteWords(), this));
```

Comment:
I think that maybe the parent should be the mpCompleter, otherwise the previous model is not deleted when setting a new.